### PR TITLE
refactor(api): prefer async/await in sms routes

### PR DESCRIPTION
**Pull request by philbooth**
Related to #2305.

@mozilla/fxa-devs r?

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/sms.js [hunk 1]
```diff
@@ -49,79 +49,51 @@ module.exports = (log, db, config, customs, sms) => {
...
+          signinCode = await db.createSigninCode(
+            sessionToken.uid,
+            metricsContext.flow_id
           );
         }
 
-        function logSuccess() {
-          return request.emitMetricsEvent(`sms.${templateName}.sent`);
-        }
+        await sms.send(phoneNumber, templateName, acceptLanguage, signinCode);
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/lib/routes/sms.js]: What happens if `request.app.features.has('signinCodes')` returns `false`, there won't be a signin code to send.

Comment (philbooth) [reply packages/fxa-auth-server/lib/routes/sms.js]: We have two links in config, one to use when there is a `signinCode` and one when there isn't it. So the SMS gets sent in both cases:

https://github.com/mozilla/fxa/blob/c27c02379581873b02d61255123f36805ee0ec1f/packages/fxa-auth-server/lib/senders/sms.js#L159-L166

Comment (shane-tomlinson) [reply packages/fxa-auth-server/lib/routes/sms.js]: Thanks for that info. Do we ever use that non-signinCode SMS anymore? Should a follow on be opened to remove it if not?

Comment (philbooth) [reply packages/fxa-auth-server/lib/routes/sms.js]: Great question! I'll have to do some research to find out the answer though.

**Pull request merged**